Reinstate ghostscript/imagick install; fix bmp name in regenerate test.

gitlost opened the pull request:
See https://github.com/wp-cli/media-command/pull/44 and https://github.com/wp-cli/media-command/pull/61

Reinstates ghostscript/imagick install which got overwritten by https://github.com/wp-cli/media-command/pull/44. This means that a number of tests weren't getting run for months :cry: 

Fixes BMP filename added in https://github.com/wp-cli/media-command/pull/61

Also adds `travis-before_install-append.yml` which will be used by PR to `scaffold-package-command` to allow for package-specific stuff to be added to `.travis.yml`.

schlessera commented:
> Also adds travis-before_install-append.yml which will be used by PR to scaffold-package-command to allow for package-specific stuff to be added to .travis.yml.

Waiting for that matching PR before merging this one.

gitlost commented:
The https://github.com/wp-cli/media-command/pull/69/commits/81a803b7a88f4bdf6574f2c767df03e262551220 commit moves the `remove_old_images()` function into `needs_regeneration()`, and splits the `$image_sizes` check in two so that thumbnails don't get destroyed if an image is skipped.

This non-destructiveness should also really extend to the no metadata case, but haven't come up with a nice way to do this, so leaving for the moment.

Also added some tests of this behaviour, and added some extra comments.

The https://github.com/wp-cli/media-command/pull/69/commits/8f7bc79947751e26f5b0fa99554b4cd1a64c8cf1 commit reinstates the caching of the ImageMagick build, adding another Travis append file.

I'll do the PR on `scaffold-package-command` for adding the extra Travis stuff very shortly...
